### PR TITLE
[1.13] Add more context to create diagnosticsts error.

### PR DESCRIPTION
**Pull request by janisz**
Backports #384
---

* Add more context to create diagnosticsts error.
* Run linter on tests
* Use go structs instead of raw JSON in list test.

Fixes: https://jira.mesosphere.com/browse/COPS-5439

### CODE REVIEW

File: .golangci.toml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+[run]
```
Comment (bamarni) [inline .golangci.toml]: Shall we switch the Makefile to use golangci? Otherwise this file is currently unused.

**Review by bamarni — APPROVED** [review]

File: .golangci.toml (hunk 1 above)
Comment (janisz) [reply .golangci.toml]: I removed this file. We can add it in a separate commit

**Comment by janisz** [comment]
:wave: @drewvanstone @br-lewis @makkes @bamarni @rgo3  Can you review, please?

**Review by bamarni — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,7 @@
 
 * Fixes
 
+  * Add better erros descriptions for `nodes diagnostics` command.
```
Comment (bamarni) [inline CHANGELOG.md]: typo here, but not blocking

**Comment by janisz** [comment]
:wave: @drewvanstone @br-lewis @makkes @bamarni @rgo3  Can you review, please?

**Review by makkes — APPROVED** [review]

**Pull request merged**